src/server/routers/tips.ts

```diff
@@ -157,4 +157,53 @@ export const tipsRouter = router({
 
       return lesson
     }),
+  markTipComplete: baseProcedure
+    .input(
+      z.object({
+        tipId: z.number(),
+      }),
+    )
+    .mutation(async ({input, ctx}) => {
+      const token = ctx?.userToken
+
+      if (!token) return null
+      const {tipId} = input
+
+      let res = await fetch(
+        `${process.env.NEXT_PUBLIC_AUTH_DOMAIN}/watch/download`,
+        {
+          method: 'POST',
+          headers: {
+            Authorization: `Bearer ${token}`,
+            'X-SITE-CLIENT': process.env.NEXT_PUBLIC_CLIENT_ID as string,
+            'Content-Type': 'application/json',
+          },
+
+          body: JSON.stringify({
+            lesson_view: {
+              lesson_id: tipId,
+            },
+          }),
+        },
+      ).then(async (res) => {
+        let json = await res.json()
+
+        return await fetch(json.complete_url, {
+          method: 'PUT',
+          headers: {
+            Authorization: `Bearer ${token}`,
+            'X-SITE-CLIENT': process.env.NEXT_PUBLIC_CLIENT_ID as string,
+            'Content-Type': 'application/json',
+          },
+
+          body: JSON.stringify({
+            id: tipId,
+          }),
+        }).then((res) => res.json())
+      })
```

Comment:
I'm chaining two calls here. `/watch/download` creates a `LessonView` that has the proper segments and then `json.complete_url` flips the `did_complete` boolean to `true.

Originally I was using the `/watch/manual_complete` endpoint but that doesn't track segments watched which is what get's instructors paid.

![image](https://github.com/skillrecordings/egghead-next/assets/6188161/e8592a45-6fbc-4061-b52c-2d0aba2a5591)